src/cli/search.rs

```diff
@@ -99,11 +100,155 @@ pub async fn execute(args: Args) -> miette::Result<()> {
+        spawn_blocking(move || {
+            let packages =
+                search_package_by_filter(&p, &repo_data, |pn, n| pn == n.as_normalized());
+            match packages {
+                Ok(packages) => Ok(packages),
+                Err(e) => Err(e),
+            }
+        }),
+    )
+    .await
+    .into_diagnostic()??;
+
+    if packages.is_empty() {
+        return Err(miette::miette!("Could not find {package_name}"));
+    }
+
+    let package = packages.last().clone();
+    if let Some(package) = package {
+        print_package_info(package);
+    }
+
+    Ok(())
+}
+
+fn print_package_info(package: &RepoDataRecord) {
+    println!();
+
+    let package = package.clone();
+    let package_name = package.package_record.name.as_source();
+    let build = &package.package_record.build;
+    let package_info = format!("{} {}", console::style(package_name), console::style(build));
+    println!("{}", package_info);
+    println!("{}\n", "-".repeat(package_info.chars().count()));
+
+    println!(
+        "{:19} {:19}",
+        console::style("Name"),
+        console::style(package_name)
+    );
+
+    println!(
+        "{:19} {:19}",
+        console::style("Version"),
+        console::style(package.package_record.version)
+    );
+
+    println!(
+        "{:19} {:19}",
+        console::style("Build"),
+        console::style(build)
+    );
+
+    let size = match package.package_record.size {
+        Some(size) => size.to_string(),
+        None => String::from("Not found."),
+    };
+    println!("{:19} {:19}", console::style("Size"), console::style(size));
+
+    let license = match package.package_record.license {
+        Some(license) => license,
+        None => String::from("Not found."),
+    };
+    println!(
+        "{:19} {:19}",
+        console::style("License"),
+        console::style(license)
+    );
+
+    println!(
+        "{:19} {:19}",
+        console::style("Subdir"),
+        console::style(package.package_record.subdir)
+    );
+
+    println!(
+        "{:19} {:19}",
+        console::style("File Name"),
+        console::style(package.file_name)
+    );
+
+    println!(
+        "{:19} {:19}",
+        console::style("URL"),
+        console::style(package.url)
+    );
+
+    let md5 = match package.package_record.md5 {
+        Some(md5) => hex::encode(md5),
+        None => "Not available".to_string(),
+    };
+    println!("{:19} {:19}", console::style("MD5"), console::style(md5));
+
+    let sha256 = match package.package_record.sha256 {
+        Some(sha256) => hex::encode(sha256),
+        None => "Not available".to_string(),
+    };
+    println!(
+        "{:19} {:19}",
+        console::style("SHA256"),
+        console::style(sha256),
+    );
+
+    println!("\nDependencies:");
+    for dependency in package.package_record.depends {
+        println!(" - {}", dependency);
+    }
+}
+
+async fn search_package_by_wildcard(
+    p: PackageName,
+    package_name: &str,
```

Comment:
Same as above with the `package_name` removal

Reply:
Renamed it but still need it for the regex pattern